contracts/src/interfaces/ITeleporterTokenBridge.sol

```diff
@@ -149,6 +150,7 @@ struct MultiHopSendMessage {
  * The source blockchain ID of the sender is known from the Teleporter message.
  */
 struct MultiHopCallMessage {
+    address originBridgeAddress;
```

Comment:
I don't think `originBridgeAddress` should be added to the `MultiHopCallMessage`. For a `MultiHopCallMessage`, the bridge address will always be the sender of the Teleporter message (given as the `originSenderAddress` in `_receivingTeleporterMessage`, and senders can't be trusted to properly set this field otherwise.

Reply:
Good call. We could still include it in this struct and defensively check that it matches the Teleporter `originSenderAddress` in `TeleporterTokenSource`, but in the case of a mismatch that would just cause the first hop to fail instead of the second. Probably not worth dedicating a field just for that.

Reply:
@michaelkaplan13 Don't we have a similar trust issue for `originSenderAddress` in `MultiHopCallMessage`? I've been trying to wrap my head around all the trust assumptions here... With our setup, destination contracts needs to trust source contracts, but not vice-versa, and a destination certainly can't trust another destination contract in a multihop. I think `originSenderAddress` can only really be trusted going from source->dest

Reply:
Ah, nevermind, that's the whole point of all of this.

Reply:
Sounds like you answered your own question, but to enumerate the cases here for clarity:

- **Destination contract sends a `SingleHopCallMessage`**
The Teleporter message is again sent to the source contract, which is also the bridge transfers destination in this case. For symmetry, the `sourceBlockchainID` and `originBridgeAddress` are specified in the message payload, but the source checks that these values match the sender of the Teleporter message when it receives it.
- **Destination contract sends a `MultiHopCallMessage`.**
Teleporter message sent by the destination contract is to the source contract, not the final destination. The `sourceBlockchainID` and `originBridgeAddress` of the `sendAndCall` message is the sender of the Teleporter message, so it doesn't need to be included in the message payload.
- **Destination contract receives a `SingleHopCallMessage`**
It can trust the `sourceBlockchainID` and `originBridgeAddress` specified in the message payload because it knows the values were set by its specified source contract, which is the only contract that it is willing to receive messages from. In the case a single hop call, the values will match the sender of the Teleporter message, but in the case of a multi hop call, the values will be of the initiating destination.
- **Destination contract receives a `MultiHopCallMessage`**
Impossible / invalid case.
- **Source contract sends a `SingleHopCallMessage`**
The source contract sets `sourceBlockchainID` and `originBridgeAddress` as its own values (i.e. `address(this)`). In this case, these values match the sender of the Teleporter message also.
- **Source contract sends a `MultiHopCallMessage`**
Impossible / invalid case.
- **Source contract receives a `SingleHopCallMessage`**
The source contract verifies that the `sourceBlockchainID` and `originBridgeAddress` in the message payload matches the sender of the Teleporter message.
- **Source contract receives a `MultiHopCallMessage`**
The source contract routes the call on to its final destination by sending a `SingleHopCallMessage`. It sets the `sourceBlockchainID` and `originBridgeAddress` in the message payload as the original sender of the Teleporter message that it received.